workers/unity/Packages/io.improbable.gdk.buildsystem/WorkerBuilder.cs

```
@@ -32,72 +39,28 @@ public static void Build()
-                            }
-                        });
-                }
+                // Parse command line arguments
+                var buildTargetFilter = CommandlineParser.GetBuildTargetFilter(args);
+                var wantedWorkerTypes = CommandlineParser.GetWorkerTypesToBuild(args);
+                var scriptImplementation = CommandlineParser.GetScriptingImplementation(args);
+                var buildEnvironment = CommandlineParser.GetBuildEnvironment(args);
 
-                var workerTypesArg = args.GetCommandLineValue(BuildWorkerTypes, "UnityClient,UnityGameLogic");
-                var wantedWorkerTypes = workerTypesArg.Split(',');
+                // Create BuildContext for each worker
+                var buildContexts = GetBuildContexts(wantedWorkerTypes, buildEnvironment, scriptImplementation,
+                    buildTargetFilter);
 
-                ScriptingImplementation scriptingBackend;
-                var wantedScriptingBackend = args.GetCommandLineValue("scriptingBackend", "mono");
-                switch (wantedScriptingBackend)
+                if (buildContexts.Count == 0)
                 {
-                    case "mono":
-                        scriptingBackend = ScriptingImplementation.Mono2x;
-                        break;
-                    case "il2cpp":
-                        scriptingBackend = ScriptingImplementation.IL2CPP;
-                        break;
-                    default:
-                        throw new BuildFailedException("Unknown scripting backend value: " + wantedScriptingBackend);
+                    throw new BuildFailedException(
+                        $"Attempted a build with no valid targets!");
                 }
 
-                var buildsSucceeded = BuildWorkers(wantedWorkerTypes, buildEnvironment, buildTargetFilter, scriptingBackend);
-
+                var buildsSucceeded = BuildWorkers(buildContexts);
```

Should this throw instead of returning a bool?